app/models/taxon_framework.rb

```diff
@@ -0,0 +1,107 @@
+    
+    upstream_taxon_frameworks = TaxonFramework.where("taxon_id IN (?)", taxon.ancestor_ids).pluck(:id)
+    ancestor_string = taxon.rank == "stateofmatter" ? taxon.id.to_s : "%/#{taxon.id}"
+    tr = TaxonFrameworkRelationship.joins(:taxa).where("taxon_framework_id IN (?) AND (taxa.id = ? OR taxa.ancestry LIKE (?) OR taxa.ancestry LIKE (?))", upstream_taxon_frameworks, taxon.id, "#{ancestor_string}", "#{ancestor_string}/%")
+    return tr.destroy_all
+  end
+  
+  def check_taxon_curators
+    return true if new_record?
+    return true if rank_level_was.nil?
+    return true unless rank_level_changed? && rank_level.nil?
+    taxon_curators.destroy_all
+    true
+  end
+  
+  def rank_level_below_taxon_rank
+    return  true if rank_level.nil?
+    if rank_level.to_i > taxon.rank_level.to_i
+      errors.add( :rank_level, "must be below the taxon rank" )
+    end
+    true
+  end
+  
+  def covers?
+    return true unless rank_level.nil?
+    return false
+  end
+
+  def get_downstream_taxon_frameworks
+    return false unless covers?
+    ancestry_string = taxon.rank == "stateofmatter" ? "#{taxon_id}" : "%/#{taxon_id}"
+    downstream_taxon_frameworks = TaxonFramework.includes("taxon").joins("JOIN taxa ON taxon_frameworks.taxon_id = taxa.id").
+      where("(taxa.ancestry LIKE ('#{ancestry_string}/%') OR taxa.ancestry LIKE ('#{ancestry_string}')) AND taxa.rank_level > #{rank_level} AND taxon_frameworks.rank_level IS NOT NULL")
+  end
+  
+  def get_unassigned_taxa
+    ancestry_string = taxon.rank == "stateofmatter" ? "#{ taxon_id }" : "#{ taxon.ancestry }/#{ taxon_id }"
+    other_taxon_frameworks = TaxonFramework.joins(:taxon).
+      where( "( taxa.ancestry LIKE ( '#{ ancestry_string }/%' ) OR taxa.ancestry LIKE ( '#{ ancestry_string }' ) )" ).
+      where( "taxa.rank_level > #{ rank_level } AND taxon_frameworks.rank_level IS NOT NULL" )
+
+    other_taxon_frameworks_taxa = ( other_taxon_frameworks.count > 0 ) ?
+      Taxon.where(id: other_taxon_frameworks.map(&:taxon_id)) : []
+
+    unassigned_taxa = Taxon.
+      joins( "JOIN taxa parent ON parent.id = (string_to_array(taxa.ancestry, '/')::int[])[array_upper(string_to_array(taxa.ancestry, '/')::int[],1)]" ).
+      where( "parent.id = #{taxon_id} OR parent.ancestry = ? OR parent.ancestry LIKE ?", ancestry_string, "#{ancestry_string}/%" ).
+      where( is_active: true ).
+      where( "parent.rank_level > ? ", rank_level).
+      where( "taxa.rank_level < ? ", rank_level).
+      where("( select count(*) from conservation_statuses ct where ct.taxon_id=taxa.id AND ct.iucn=70 AND ct.place_id IS NULL ) = 0")
+
+    other_taxon_frameworks_taxa.each do |t|
+      unassigned_taxa = unassigned_taxa.where("parent.ancestry != ? AND parent.ancestry NOT LIKE ?", "#{t.ancestry}/#{t.id}", "#{t.ancestry}/#{t.id}/%")
+    end
+
+    return unassigned_taxa
+  end
+  
+  def get_flagged_taxa
+    flagged_taxa = Taxon.get_internal_taxa_covered_by(self).
```

Comment:
If you're just returning a scope and not assigning results to some kind of instance variable, there's no need to assign it to a local variable.